From f01498ec4db65a626df1906b58446a2f6dbc2b26 Mon Sep 17 00:00:00 2001
From: Dmytro Firsov <dmytro_firsov@epam.com>
Date: Mon, 2 Dec 2024 18:25:49 +0200
Subject: [PATCH 1/2] boards: xenvm: return separate defconfig for xenvm with
 GICv3

During migration to HWMv2 separate specific defconfig file was removed
for GICv3 version of virtual xenvm boards. It worked fine before
commit 0be0d2175bdb ("cmake: modules: extensions: Revert using common
board files") significantly changed build behavior, but did not return
previously removed file. This led to build/runtime issues, when some of
the Kconfig options were not selected.

Return GICv3 specific defconfig to board directory to fix configuration
problems.

Signed-off-by: Dmytro Firsov <dmytro_firsov@epam.com>
---
 boards/xen/xenvm/xenvm_xenvm_gicv3_defconfig | 13 +++++++++++++
 1 file changed, 13 insertions(+)
 create mode 100644 boards/xen/xenvm/xenvm_xenvm_gicv3_defconfig

diff --git a/boards/xen/xenvm/xenvm_xenvm_gicv3_defconfig b/boards/xen/xenvm/xenvm_xenvm_gicv3_defconfig
new file mode 100644
index 00000000000..2115f9175bd
--- /dev/null
+++ b/boards/xen/xenvm/xenvm_xenvm_gicv3_defconfig
@@ -0,0 +1,13 @@
+# Enable UART driver
+CONFIG_SERIAL=y
+
+CONFIG_MAX_XLAT_TABLES=24
+
+# Enable console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+# Enable logging subsys
+CONFIG_LOG=y
+CONFIG_LOG_MODE_MINIMAL=n
+CONFIG_TIMER_READS_ITS_FREQUENCY_AT_RUNTIME=y
-- 
2.34.1

